src/wallet/wallet.cpp

```diff
@@ -3776,27 +3821,24 @@ bool CWallet::CreateCoinStake(ChainstateManager& chainman, const CWallet* pwalle
             return error("%s() : deserialize or I/O error in CreateCoinStake()", __PRETTY_FUNCTION__);
         }
 
-
         // Attempt to add more inputs
         // Only add coins of the same key/address as kernel
-        if (txNew.vout.size() == 2 && ((pcoin->txout.scriptPubKey == scriptPubKeyKernel || pcoin->txout.scriptPubKey == txNew.vout[1].scriptPubKey))
+        if (((pcoin->txout.scriptPubKey == scriptPubKeyKernel || pcoin->txout.scriptPubKey == txNew.vout[1].scriptPubKey))
             && pcoin->outpoint.hash != txNew.vin[0].prevout.hash)
         {
-            // Stop adding more inputs if already too many inputs
-            if (txNew.vin.size() >= 100)
+            // Stop adding more inputs if already too many inputs and we are above target
+            if ((txNew.vin.size() >= MAX_COINSTAKE_INPUTS) && (nCredit > nTargetOutputAmount))
                 break;
-            // Stop adding more inputs if value is already pretty significant
-            if (nCredit > nCombineThreshold)
+            // Stop adding more inputs if we are below target but about to exceed 1kb
+            if ((txNew.vin.size() >= 8) && (nCredit < nTargetOutputAmount))
```

Comment:
Do 8 maximally sized inputs fit when there is the minter key output in addition to the other output?

Reply:
good catch about minter key, i've added the condition during adding inputs